### PR TITLE
34952 fixed equals inconsistencies

### PR DESCRIPTION
**Pull request by theo-ritense**
- added type field to equals comparison
- removed singular nextStep from configuration
- made some columns non-nullable to keep code cleaner

### CODE REVIEW

File: form-flow/src/main/resources/config/form-flow/schema/formflow.schema.json [hunk 1]
```diff
@@ -75,10 +75,6 @@
                                 "step"
                             ]
                         }
-                    },
-                    "nextStep:": {
-                        "type": "string",
-                        "description": "The key of the next step in the flow. When multiple steps should be available you should use the property 'nextSteps'"
                     }
                 },
                 "required": [
```
Comment (Rick-Ritense) [inline form-flow/src/main/resources/config/form-flow/schema/formflow.schema.json:80]: You can now make nextSteps required right?

Comment (theo-ritense) [reply form-flow/src/main/resources/config/form-flow/schema/formflow.schema.json:80]: ah, yes. Will do

Comment (theo-ritense) [reply form-flow/src/main/resources/config/form-flow/schema/formflow.schema.json:80]: I did say yes at first, but a final step will not have next steps.

Comment (Rick-Ritense) [reply form-flow/src/main/resources/config/form-flow/schema/formflow.schema.json:80]: ah yeah of course

**Review by Rick-Ritense — APPROVED** [review]

**Pull request merged**